google-cloud-storage/test/helper.rb

```diff
@@ -71,14 +71,64 @@ def random_bucket_hash name: random_bucket_name,
                          enable_object_retention: nil,
                          effective_time: DateTime.now,
                          retention_duration_seconds: 604800, # 7 days
-                         hierarchical_namespace: nil
+                         hierarchical_namespace: nil,
+                         generation: "1733393981548601746"
     versioning_config = { "enabled" => versioning } if versioning
     { "kind" => "storage#bucket",
       "id" => name,
       "selfLink" => "#{url_root}/b/#{name}",
       "projectNumber" => "1234567890",
       "name" => name,
       "timeCreated" => Time.now,
+      "generation" => generation,
+      "metageneration" => "1",
+      "owner" => { "entity" => "project-owners-1234567890" },
+      "location" => location,
+      "locationType" => location_type,
+      "rpo" => rpo,
+      "cors" => cors,
+      "lifecycle" => lifecycle,
+      "logging" => logging_hash(logging_bucket, logging_prefix),
+      "storageClass" => storage_class,
+      "versioning" => versioning_config,
+      "website" => website_hash(website_main, website_404),
+      "billing" => billing_hash(requester_pays),
+      "etag" => "CAE=",
+      "autoclass" => autoclass_config_hash(autoclass_enabled, autoclass_terminal_storage_class),
+      "enableObjectRetention" => enable_object_retention,
+      "softDeletePolicy" => soft_delete_policy_object(retention_duration_seconds: retention_duration_seconds),
+      "hierarchicalNamespace" => hierarchical_namespace
+    }.delete_if { |_, v| v.nil? }
+  end
+  def random_deleted_bucket_hash name: random_bucket_name,
```

Comment:
I think we can reuse the `random_bucket_hash` for getting a deleted bucket hash. We will have to modify the method to incorporate soft_delete related changes.